Unable to compile tensorflow in Win/Debug mode

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows Server 2012
- **TensorFlow installed from (source or binary)**:
Source

- **TensorFlow version (use command below)**:
92838685241ca22ea2797e937e380bdbe8325784

- **Python version**: 
3.5.4
- **Bazel version (if compiling from source)**:

- **GCC/Compiler version (if compiling from source)**:
VS 2015

- **CUDA/cuDNN version**:
None
- **GPU model and memory**:
None
- **Exact command to reproduce**:
"D:\src\tensorflow\b\packages\cmake\bin\cmake" -G "Visual Studio 14 2015 Win64" D:\src\tensorflow\s\tensorflow\contrib\cmake -DCMAKE_BUILD_TYPE=Debug -DPYTHON_EXECUTABLE="D:\src\tensorflow\b\packages\python\python.exe" -DPYTHON_LIBRARIES="D:\src\tensorflow\b\packages\python\libs\python35.lib" -DSWIG_EXECUTABLE=D:\src\tensorflow\b\packages\swigwin.3.0.9\tools\swigwin-3.0.9\swig.exe -Dtensorflow_BUILD_SHARED_LIB=ON -Dtensorflow_BUILD_PYTHON_TESTS=OFF 

Then build tensorflow.vcxproj

### Describe the problem
build failed

### Source code / logs
```
 ##[error]C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(958,0): Error C2678: binary '<': no operator found which takes a left-hand operand of type 'tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator' (or there is no acceptable conversion) (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
     11>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(958): error C2678: binary '<': no operator found which takes a left-hand operand of type 'tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator' (or there is no acceptable conversion) (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc) [D:\src\tensorflow\b\debug\tf_core_kernels.vcxproj]
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\thread(224): note: could be 'bool std::operator <(std::thread::id,std::thread::id) noexcept' (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\system_error(436): note: or       'bool std::operator <(const std::error_condition &,const std::error_condition &) noexcept' (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\system_error(427): note: or       'bool std::operator <(const std::error_code &,const std::error_code &) noexcept' (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(958): note: while trying to match the argument list '(tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator, tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator)' (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xutility(967): note: see reference to function template instantiation 'void std::_Debug_range2<_InIt>(_RanIt,_RanIt,std::_Dbfile_t,std::_Dbline_t,std::random_access_iterator_tag)' being compiled
                  with
                  [
                      _InIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator,
                      _RanIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator
                  ] (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\algorithm(2196): note: see reference to function template instantiation 'void std::_Debug_range<_FwdIt>(_InIt,_InIt,std::_Dbfile_t,std::_Dbline_t)' being compiled
                  with
                  [
                      _FwdIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator,
                      _InIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator
                  ] (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\algorithm(2206): note: see reference to function template instantiation '_FwdIt std::lower_bound<_FwdIt,_Ty,std::less<void>>(_FwdIt,_FwdIt,const _Ty &,_Pr)' being compiled
                  with
                  [
                      _FwdIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator,
                      _Ty=tensorflow::int64,
                      _Pr=std::less<void>
                  ] (compiling source file D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc)
          D:\src\tensorflow\s\tensorflow\contrib\boosted_trees\lib\utils\sparse_column_iterable.cc(115): note: see reference to function template instantiation '_FwdIt std::lower_bound<tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator,tensorflow::int64>(_FwdIt,_FwdIt,const _Ty &)' being compiled
                  with
                  [
                      _FwdIt=tensorflow::boosted_trees::utils::`anonymous-namespace'::IndicesRowIterator,
                      _Ty=tensorflow::int64
                  ]
```

--- a/tensorflow/contrib/boosted_trees/lib/utils/sparse_column_iterable.cc
+++ b/tensorflow/contrib/boosted_trees/lib/utils/sparse_column_iterable.cc
@@ -96,6 +96,10 @@ class IndicesRowIterator
     return (row_idx_ != other.row_idx_);
   }
 
+  bool operator<(const IndicesRowIterator& other) const {
+	return (row_idx_ < other.row_idx_);
+  }
+
   bool operator==(const IndicesRowIterator& other) const {
     QCHECK_EQ(iter_, other.iter_);
     return (row_idx_ == other.row_idx_);

